packages/utils/provider-amt/build.yaml

```diff
@@ -0,0 +1,18 @@
+image: busybox
+unpack: true
+copy:
+  - image: "{{ .Values.labels.repository }}:v{{ .Values.version }}"
+    source: /usr/local/lib/librpc.so
+    destination: /usr/local/lib/librpc.so
+  - image: "{{ .Values.labels.repository }}:v{{ .Values.version }}"
+    source: /usr/local/include/librpc.h
+    destination: /usr/local/include/librpc.h
+  - image: "{{ .Values.labels.repository }}:v{{ .Values.version }}"
+    source: /system/providers/agent-provider-amt
+    destination: /usr/local/system/providers/agent-provider-amt
+steps:
+  - chmod +x /usr/local/system/providers/agent-provider-amt
+includes:
+  - /usr/local/lib/librpc.so
```

Comment:
This seems to be becuase on our provider we link to C libraries directly. Why is that @paynejacob ? I saw that intel provides a pure go client for atm at https://github.com/open-amt-cloud-toolkit/rpc-go which directly uses the rpc stuff in there with no linking. 

Maybe we can use that so we also have a pure go lib for this?

*Disclaimer*: Is like the first time I heard about amt so no idea of the drawback of using go or how it even works, so maybe my comment is wrong!

Reply:
That was my first approach 😄  to just use their library but all the functionality we needed was in the internal package so it wasn't accessible.  The alternative would be to make a subprocess call to their rpc binary but imo subprocess calls are fragile.  I noticed in their docs you could build the library as a c library and it allowed for making rpc calls directly.  This was the cleanest solution I could find.  I am open to revisiting any of these, given we are pivoting to a bundle I think the easiest thing would be to tell our binary to search it's current directory for the lib and include the library in the bundle.  We could also use the subprocess call to their binary but I think that is about the same complexity.

Reply:
ahhh...go and its stupid internal dir ¬_¬

Maybe we can open an issue on the repo so they can extract that to a different dir so its available as a lib for other projects? Or, as its apache, we can just bring the whole internal dir into our repo directly as long as we keep the apache license :D

Reply:
Ill open that issue, I agree just importing their code would be ideal.  For now I have it working as a bundle with the library and binary.  The binary checks for the library in it's current directory so it will be packageable as a bundle.  I think it will be easier than trying to maintain a fork for now.